### ISSUE TITLE
Login on a newly installed ICAT raises InternalError if lucene directory is empty

### ISSUE THREAD

**Issue by RKrahl** [icatproject/icat.server#151]
I got an InternalError when trying to log in to a newly installed ICAT if the lucene directory is empty.

Steps to reproduce:
1. Start with a new glassfish.
2. Deploy authenticators.
3. Make sure the lucene directory configured in the next step is empty.
4. Deploy icat.server.
5. Log in.

Symptom:
```
>>> client.login(conf.auth, conf.credentials)
ERROR:suds.client:<suds.sax.document.Document instance at 0x22b2e18>
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python2.7/site-packages/icat/client.py", line 343, in login
    raise translateError(e)
icat.exception.ICATInternalError: javax.ejb.EJBException: javax.ejb.EJBException: javax.ejb.CreateException: Could not create stateless EJB
        at com.sun.ejb.containers.StatelessSessionContainer._getContext(StatelessSessionContainer.java:435)
        at ...
```
See the [server.log](https://gist.github.com/RKrahl/e4b9b302724a919645b3#file-icat-server_151-server-log) for the full error trace.

The problem vanishes if glassfish is restarted once after the error is raised.

**Comment by RKrahl** [comment]
According to the trace, the error is triggered by the commit and reopen in LuceneSingleton.java:
```java
	@PostConstruct
	private void init() {

		luceneDirectory = propertyHandler.getLuceneDirectory();
		active = luceneDirectory != null;
		if (active) {
			luceneRefreshSeconds = propertyHandler.getLuceneRefreshSeconds() * 1000L;
			luceneCommitCount = propertyHandler.getLuceneCommitCount();
			ei = EntityInfoHandler.getInstance();

			try {
				directory = FSDirectory.open(new File(luceneDirectory));
				logger.debug("Opened FSDirectory with lockid " + directory.getLockID());
				Analyzer analyzer = new IcatAnalyzer();
				IndexWriterConfig config = new IndexWriterConfig(Version.LATEST, analyzer);
				iwriter = new IndexWriter(directory, config);
				String[] files = directory.listAll();
				if (files.length == 1 && files[0].equals("write.lock")) {
					logger.debug("Directory only has the write.lock file so commit and reopen");
					iwriter.commit();
					iwriter.close();
					iwriter = new IndexWriter(directory, config);
				}
```

This explains that the error only occurs if the lucene directory is empty. The commit leaves two files in the lucene directory, this explains why restarting glassfish cures the problem.

**Comment by fisherab** [comment]
Thanks for this. It might explain why the icat installer always fails first time.

**Comment by fisherab** [comment]
I have decided to fix this in a 4.5.1 release rather than waiting for 4.6.
